contracts/Strategy.sol

```diff
@@ -376,10 +460,18 @@ contract Strategy is BaseStrategy {
         }
     }
 
+    function delegatedAssets() external view override returns (uint256) {
+        // returns total debt borrowed in want (which is the delegatedAssets)
+        return
+            _fromETH(
+                _toETH(balanceOfDebt(), address(investmentToken)),
+                address(want)
+            );
+    }
+
     function prepareMigration(address _newStrategy) internal override {
-        // in yearn-vaults, the oldStrategy's totalDebt is set to 0 before calling migrate on BaseStrategy
-        // so we need to use totalDebt of the _newStrategy even if that logic "does not make sense"
-        liquidatePosition(vault.strategies(_newStrategy).totalDebt);
+        // not implemented because debt cannot be migrated
+        revert();
```

Comment:
this is the first time i ever seen  this in a strategy logic, code for prepareMigration should try its best to unwind debt,

Reply:
I talked about this with @fubuloubu. Since debt can't be migrated, it doesn't make sense to migrate the strategy.

To do a migration we would need to:
 - revoke
 - harvest
 - clone anew strat

Perhaps we should leave the migrate empty instead of revert(), 🤷 
wdyt @fubuloubu?

Reply:
we added revert to be sure the user / management knows that this is not migrating the strategy

Reply:
Left my thinking here,

https://github.com/yearn/yearn-strategies/issues/83#issuecomment-847318794

We should try to keep the more plans/options available to recover funds, migrate is one option